NitroxPatcher/Patches/Dynamic/SeaDragonMeleeAttack_OnTouchFront_Patch.cs

```diff
@@ -0,0 +1,117 @@
+using System.Collections.Generic;
+using System.Reflection;
+using System.Reflection.Emit;
+using HarmonyLib;
+using NitroxClient.Communication.Abstract;
+using NitroxClient.GameLogic;
+using NitroxClient.GameLogic.PlayerLogic;
+using NitroxModel.DataStructures;
+using NitroxModel.Helper;
+using NitroxModel.Packets;
+using UnityEngine;
+
+namespace NitroxPatcher.Patches.Dynamic;
+
+/// <summary>
+/// Prevents non simulating players from running locally <see cref="SeaDragonMeleeAttack.OnTouchFront"/>.
+/// Broadcasts for the simulating player:
+/// - attack on cyclops effects
+/// - attack on local player effects
+/// - attack on remote players effects
+/// </summary>
+public sealed partial class SeaDragonMeleeAttack_OnTouchFront_Patch : NitroxPatch, IDynamicPatch
+{
+    internal static readonly MethodInfo TARGET_METHOD = Reflect.Method((SeaDragonMeleeAttack t) => t.OnTouchFront(default));
+
+    public static bool Prefix(SeaDragonMeleeAttack __instance)
+    {
+        if (!__instance.TryGetNitroxId(out NitroxId creatureId) ||
+            Resolve<SimulationOwnership>().HasAnyLockType(creatureId))
+        {
+            return true;
+        }
+
+        return false;
+    }
+
+    /*
+     * 1st injection:
+     * base.gameObject.SendMessage("OnMeleeAttack", target, SendMessageOptions.DontRequireReceiver);
+     * this.timeLastBite = Time.time;
+     * BroadcastSeaDragonAttackTarget(this, target);           <---- INSERTED LINE
+     * 
+     * 2nd injection:
+     *     global::Utils.PlayEnvSound(this.attackSound, collider.transform.position, 20f);
+     * }
+     * component3.gameObject.GetComponent<LiveMixin>().TakeDamage(this.biteDamage, default(Vector3), DamageType.Normal, base.gameObject);
+     * BroadcastSeaDragonAttackTarget(this, target);       <---- INSERTED LINE
+     * 
+     * 3rd injection:
+     * Exosuit component4 = target.GetComponent<Exosuit>();
+     * BroadcastSeaDragonAttackRemotePlayer(this, target);            <---- INSERTED LINE
+     * if (component4 != null)
```

Comment:
This seems a little to early, no? Broadcasting before all checks are done if the exosuit can even be grabed

Reply:
At this point, all that counts is whether or not ``component4`` is null, which I check in ``BroadcastSeaDragonAttackRemotePlayer`` so the broadcast is fine at this place